### PR TITLE
Add support for empty source packages

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
Fixes https://bugs.swift.org/browse/SR-793

I am not sure if rootdir should be passed to transmute method, maybe Package should have a property isRoot ?

### CODE REVIEW

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by aciidb0mb3r** [comment]
@mxcl should I update anything?

File: Sources/Transmute/transmute().swift [hunk 1]
```diff
@@ -10,14 +10,25 @@
...
 
     for package in packages {
-        let modules = try package.modules()
+
+        let modules: [Module]
+        do {
+            modules = try package.modules()
+        } catch Package.ModuleError.NoModules(let pkg) where pkg.path == rootdir {
+            //Ignore and print warning if root package doesn't contain any sources
+            print("warning: root package '\(pkg)' does not contain any sources")
```
Comment (kostiakoval) [inline Sources/Transmute/transmute().swift]: Would be great to handle warning in similar way to errors.  
That way we could hold warnings in one place and make them statically typed, by using enums.
Also we could use different text color for warnings as well. 
But maybe it's out of the scope for that PR.

Comment (aciidb0mb3r) [inline Sources/Transmute/transmute().swift]: Yeah, agreed. I was actually thinking to do that but thought it'd be too much for this PR. I would love to take a shot at that in a sep. PR

**Comment by aciidb0mb3r** [comment]
rebased

**Comment by aciidb0mb3r** [comment]
@mxcl trigger CI please

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by aciidb0mb3r** [comment]
I was just going to comment that merging this would create conflict in #174

**Comment by mxcl** [comment]
That's OK I can resolve my own PR conflicts.

**Comment by aciidb0mb3r** [comment]
😂Yeah just wasn't sure if its ok for that and swift-3-api to be conflicted as they're major things right now as you said 🖖

**Comment by mxcl** [comment]
Oh indeed, but as long as we're on top of it, the vast Swift machine will keep turning. I'm fixing up those branches as we speak.

**Pull request merged**